pkgs/applications/editors/vscode/extensions/rust-lang.rust-analyzer/default.nix

```diff
@@ -21,55 +22,50 @@ let
   # Use the plugin version as in vscode marketplace, updated by update script.
   inherit (vsix) version;
 
-  releaseTag = "2024-07-08";
+  releaseTag = "2025-02-17";
 
   src = fetchFromGitHub {
     owner = "rust-lang";
     repo = "rust-analyzer";
     rev = releaseTag;
-    hash = "sha256-STmaV9Zu74QtkGGrbr9uMhskwagfCjJqOAYapXabiuk=";
+    hash = "sha256-i76MMFSkCr4kDwurK8CACwZq7qEgVEgIzkOr2kiuAKk=";
   };
 
-  build-deps =
-    nodePackages."rust-analyzer-build-deps-../../applications/editors/vscode/extensions/rust-lang.rust-analyzer/build-deps";
-  # FIXME: Making a new derivation to link `node_modules` and run `npm run package`
-  # will cause a build failure.
-  vsix = build-deps.override {
+  vsix = buildNpmPackage {
+    inherit pname releaseTag;
+    version = lib.trim (lib.readFile ./version.txt);
     src = "${src}/editors/code";
-    outputs = [
-      "vsix"
-      "out"
+    npmDepsHash = "sha256-0frOGphtzO6z8neSEYfjyRYrM6zEO3wId/TACblZkxM=";
+    buildInputs = [
+      pkgsBuildBuild.libsecret
     ];
```

Comment:
There *needs* to be a better way to do this

Without it, the npm configure phase cannot find libsecret at all (as the `PKG_CONFIG_PATH` isn’t set up correctly) and the soonest hook i can change the PKG_CONFIG_PATH at appears to be *before* `PKG_CONFIG_PATH_HOST` is set.

Hardcoding PKG_CONFIG_PATH is an option but that seems like even more of a hack.

The change between this and the previous revision of this PR is that this should work when cross-compiling [i am testing it out right now]